presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -171,6 +171,15 @@ default void createTable(ConnectorSession session, ConnectorTableMetadata tableM
     }
 
     /**
+     * Creates a table using the specified table metadata. If the table already exists and
+     * failIfExists is true this method should throw an exception.
+     */
+    default void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean failIfExists)
```

Comment:
We should add this parameter to the existing method rather than adding a new method. I also think this should be `ignoreExisting`, since the normal behavior is to throw an exception. (saying `IF NOT EXISTS` / `ignoreExisting` is a special case)

Let's also add the same for `dropTable`, so we can make the incompatible changes at the same time.